### ISSUE TITLE
Scaffold-DbContext failed while using C# 7.0 above (exclude 7.0)

### ISSUE THREAD

**Issue by hez2010** [aspnet/EntityFrameworkCore#11981]
```
Exception message: build failed
```

### Steps to reproduce
dotnet new mvc --auth Individual
use identity scaffolder to scaffold identity code
edit csproj and add
```xml
<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
  <LangVersion>latest</LangVersion>
</PropertyGroup>
<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
  <LangVersion>latest</LangVersion>
</PropertyGroup>
```
then edit Program.cs:
change Main method to 
```cs
public async static Task Main(string[] args)
{
    await CreateWebHostBuilder(args).Build().RunAsync();
}
```
and build it to make sure it can be built successfully.
then use Scaffold-DbContext or dotnet ef dbcontext, it shows "build failed"

ALSO, IF YOU CHANGE THE MAIN METHOD TO ASYNC METHOD BEFORE SCAFFOLD IDENTITY, THE IDENTITY SCAFFOLDER WILL FAIL AND SHOWS:
![untitled](https://user-images.githubusercontent.com/14960345/40037512-36e31fee-5840-11e8-86e8-967aa20cfc5b.png)

What's more, I found a new scaffolder bug that, it will keep original namespace the same as folder name "New folder" but not "New_folder", which means the generated code can not be compiled successfully unless user replace the blank space with "_" manually.

### Further technical details
EF Core version: 2.1 rc1 (found in project.csproj or packages.config)
Database Provider: Microsoft.EntityFrameworkCore.Sqlite (e.g. Microsoft.EntityFrameworkCore.SqlServer)
Operating system: Windows 10 1803 x64
IDE: Visual Studio 2017 15.7 (e.g. Visual Studio 2017 15.4)

**Comment by hez2010** [comment]
@bricelam I've updated the steps to repo this problem

**Comment by AndyCJ** [comment]
I hit this today too.

I also found that if you've moved the SqlDbContext in to another project so it can be referenced by other projects then it can't find the SqlDbContext in the drop down.

**Comment by ajcvickers** [comment]
@hez2010 @AndyCJ The issue was reported on a 2.1 RC release. Does this still happen with the latest 2.1 releases of the SDK and EF Core?

**Comment by hez2010** [comment]
@ajcvickers This problem still exists when scaffolding identity pages in .NET Core SDK 2.1.403. But Scaffold-DbContext and dotnet ef dbcontext works fine.
And also I found that the scaffolded identity pages miss a "ShowRecoveryCodes" page.

**Comment by AndyCJ** [comment]
I gave it a quick test and it seems fine for me now. I'm not sure what I messed up. I must have been on an old version as you say.

1. Created new mvc project via dotnet "dotnet new mvc --auth Individual" (.NET Command Line Tools (2.1.402)).
2. Changed project language version to "latest minor version" via VS (15.8.6).
3. Restored NuGet packages via VS.
4. Updated NuGet packages via VS.
5. Changed Program.cs to have "public static async Task Main(string[] args)" as the entry point.
6. Ran "Scaffold-DbContext -connection "DataSource=app.db" Microsoft.EntityFrameworkCore.Sqlite" via the package manager console.
